src/Features/CSharp/Portable/Snippets/AbstractCSharpForLoopSnippetProvider.cs

```diff
@@ -48,19 +47,42 @@ protected override SyntaxNode GenerateStatement(SyntaxGenerator generator, Synta
 
             TypeSyntax? iteratorTypeSyntax = null;
 
+            var inlineExpression = inlineExpressionInfo?.Node.WithoutLeadingTrivia();
```

Comment:
could you extract this all to a helper local function?

Reply:
specifically, i really hate having to look ata ll the flow paths and see how they interact with each other :)